native-apps/unity_android_deeplink/ii_integration_page/src/greet_frontend/src/index.js

```diff
@@ -0,0 +1,83 @@
+import {AuthClient} from "@dfinity/auth-client"
+import {SignIdentity} from "@dfinity/agent";
+import {DelegationIdentity, Ed25519PublicKey } from "@dfinity/identity";
+
+// An incomplete Ed25519KeyIdentity with only the public key provided.
+class IncompleteEd25519KeyIdentity extends SignIdentity {
+    constructor(publicKey) {
+        super();
+        this._publicKey = publicKey;
+    }
+
+    getPublicKey () {
+        return this._publicKey;
+    }
+}
+
+function fromHexString(hexString) {
+    return new Uint8Array((hexString.match(/.{1,2}/g) ?? []).map(byte => parseInt(byte, 16))).buffer;
+}
+
+let myKeyIdentity;
+let sessionKeyIndex = -1;
+
+var url = window.location.href;
+sessionKeyIndex = url.indexOf("sessionkey=");
+if (sessionKeyIndex !== -1) {
+    // Parse the public session key and instantiate an IncompleteEd25519KeyIdentity.
+    var sessionkey = url.substring(sessionKeyIndex + "sessionkey=".length);
+
+    var publicKey = Ed25519PublicKey.fromDer(fromHexString(sessionkey));
+    myKeyIdentity = new IncompleteEd25519KeyIdentity(publicKey);
+} else {
+    // TODO: initialize an Ed25519KeyIdentity();
+}
+
+let delegationIdentity;
+
+const loginButton = document.getElementById("login");
+loginButton.onclick = async (e) => {
+    e.preventDefault();
+
+    // Create an auth client.
+    let authClient = await AuthClient.create({
+        identity: myKeyIdentity,
+    });
+
+    // Start the login process and wait for it to finish.
+    await new Promise((resolve) => {
+        authClient.login({
+            identityProvider: "https://identity.ic0.app/#authorize",
+            onSuccess: resolve,
+        });
+    });
+
+    // At this point we're authenticated, and we can get the identity from the auth client.
+    const identity = authClient.getIdentity();
+    if (identity instanceof DelegationIdentity) {
+        delegationIdentity = identity;
+    }
+
+    return false;
+};
+
+const openButton = document.getElementById("open");
+openButton.onclick = async (e) => {
+    e.preventDefault();
+
+    if (sessionKeyIndex === -1) {
+        // TODO: warning for not login from a game.
+        return false;
+    }
+    
+    var url = "internetidentity://authorize?";
```

Comment:
This is an app-link, correct?

Reply:
This is a deep link, you can customize the scheme as you wish.

And for app links, per https://developer.android.com/training/app-links/, you can only use HTTP or HTTPS schemes.

Reply:
See other comment, I did not realize there are two types of integration. Perfect 🙂